rest/src/main/java/com/redhat/ipaas/rest/v1/dao/DataManager.java

```diff
@@ -257,7 +266,7 @@ public boolean delete(String kind, String id) {
             throw new EntityNotFoundException("Can not find " + kind + " with id " + id);
 
         WithId entity = cache.get(id);
-        if (entity != null && doWithDataAccessObject(kind, d -> d.delete(entity))) {
+        if (doWithDataAccessObject(kind, d -> d.delete(entity)) || entity != null ) {
```

Comment:
But in https://github.com/rhuss/ipaas-rest/blob/6f00266a4ddef2644ab34d6b7b737fbf74981baa/rest/src/main/java/com/redhat/ipaas/rest/v1/dao/RealTimeAbstractDAO.java#L129 when entity is `null` in this closure it will throw an exception or do I miss sth ?